Applications/DataExplorer/DataView/MeshElementRemovalDialog.cpp

```diff
@@ -78,13 +77,21 @@ void MeshElementRemovalDialog::accept()
 	}
 	if (this->boundingBoxCheckBox->isChecked())
 	{
-		const MeshLib::Node min(this->xMinEdit->text().toDouble(),
-		                        this->yMinEdit->text().toDouble(),
-		                        this->zMinEdit->text().toDouble(),0);
-		const MeshLib::Node max(this->xMaxEdit->text().toDouble(),
-		                        this->yMaxEdit->text().toDouble(),
-		                        this->zMaxEdit->text().toDouble(),0);
-		ex.searchByBoundingBox(min, max);
+		std::vector<MeshLib::Node*> const& nodes (_project.getMesh(this->meshNameComboBox->currentText().toStdString())->getNodes());
+		GeoLib::AABB<MeshLib::Node> aabb(nodes.begin(), nodes.end());
+		MeshLib::Node minAABB = aabb.getMinPoint();
+		MeshLib::Node maxAABB = aabb.getMaxPoint();
+		double const eps (std::numeric_limits<double>::epsilon());
+
+		// only extract bounding box parameters that have been edited (otherwise there will be rounding errors!)
+		minAABB[0] = (aabb_edits[0]) ? this->xMinEdit->text().toDouble() : (minAABB[0] - eps);
```

Comment:
when checking if a point p is contained within an ```AABB```, the method returns true if p is "larger" (>) than the min-point and "smaller" (<) than the max-point. i.e. if another point is located on the boundary, it is per definition *not* contained. to make sure such points are not removed, eps is subtracted or added, respectively.